files/class-api-client.php

```diff
@@ -173,12 +180,10 @@ public function get_file( $file_path ) {
 			return new WP_Error( 'get_file-failed', sprintf( __( 'Failed to get file `%1$s` (response code: %2$d)' ), $file_path, $response_code ) );
 		}
 
-		$body = wp_remote_retrieve_body( $response );
-
 		// save to cache
-		$this->cache->cache_file( $file_path, $body );
+		$this->cache->cache_file( $file_path, $tmp_file );
 
-		return $body;
+		return fopen( $tmp_file, 'r+' );
```

Comment:
Should this be `$this->cache->get_file( $file_path )`? I worry a bit about duplicating the details of how we're reading temp files from disk.

Reply:
Agreed about duplication and I think we should move the `fopen()` calls to the Stream wrapper class instead so that it's opened with the correct mode.

Reply:
Changed in 77af3de4b107b9a46476a537f6eb42e503987769